### ISSUE TITLE
Error while processing tensorflow/tensorflow repository

### ISSUE THREAD

**Issue by Dim25** [sourcecred/sourcecred#1023]
I was able to run `node bin/sourcecred.js load sourcecred/sourcecred` successfully, but when I try to process the [tensorflow](https://github.com/tensorflow/tensorflow) repository with `node bin/sourcecred.js load tensorflow/tensorflow` I get the following error:

> Starting tasks
>   GO   load-git
>   GO   load-github
>  DONE  load-git
>  FAIL  load-github
> Exit code: 1
> Contents of stderr:
>     Error: Cannot update data for nonexistent node: undefined
>         at Mirror._nontransactionallyUpdateOwnData (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:1081:415)
>         at Mirror._nontransactionallyUpdateData (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:885:489)
>         at _inTransaction (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:890:605)
>         at _inTransaction (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:1324:142)
>         at Mirror._updateStep (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:890:522)
>         at process.internalTickCallback (internal/process/next_tick.js:77:7)
> 
> 
> Full results
>  DONE  load-git
>  FAIL  load-github
> Exit code: 1
> Contents of stderr:
>     Error: Cannot update data for nonexistent node: undefined
>         at Mirror._nontransactionallyUpdateOwnData (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:1081:415)
>         at Mirror._nontransactionallyUpdateData (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:885:489)
>         at _inTransaction (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:890:605)
>         at _inTransaction (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:1324:142)
>         at Mirror._updateStep (/home/diver/in_work/sourcecred/sourcecred/bin/sourcecred.js:890:522)
>         at process.internalTickCallback (internal/process/next_tick.js:77:7)
> 
> 
> Overview
> Failed tasks:
>   - load-github
> Final result:  FAILURE 
> 

Node v11.1.0
Yarn 1.10.1
Ubuntu 18.04.1 LTS x86_64

**Comment by Dim25** [comment]
Same error with `node bin/sourcecred.js load keras-team/keras`

**Comment by decentralion** [comment]
While attempting to repro (on TensorFlow) I got the following _different_ error. I suspect it may be ephemeral and am going to retry, but posting it for posterity:

---

    Unexpected GraphQL error; this may be a bug in SourceCred:  {"postBody":"{\"query\":\"query MirrorUpdate { owndata_0: nodes(ids: [ \\\"MDEyOk9yZ2FuaXphdGlvbjE1NjU4NjM4\\\" ]) { ... on Organization { __typename id url login } } owndata_1: nodes(ids: [ \\\"MDM6UmVmNDU3MTcyNTA6bWFzdGVy\\\" ]) { ... on Ref { __typename id target { __typename ... on Blob { id } ... on Commit { id } ... on Tag { id } ... on Tree { id } } } } owndata_2: nodes(ids: [ \\\"MDU6SXNzdWUxMTU4ODYzMDI=\\\" \\\"MDU6SXNzdWUxMTU4OTQxMzg=\\\" \\\"MDU6SXNzdWUxMTU4OTY2NTY=\\\" \\\"MDU6SXNzdWUxMTU4OTg0NDk=\\\" \\\"MDU6SXNzdWUxMTU5MTA5MDA=\\\" \\\"MDU6SXNzdWUxMTU5MTE3MzU=\\\" \\\"MDU6SXNzdWUxMTU5MTIxMTE=\\\" \\\"MDU6SXNzdWUxMTU5MTQ0OTU=\\\" \\\"MDU6SXNzdWUxMTU5MTQ2OTE=\\\" \\\"MDU6SXNzdWUxMTU5MTQ3NjY=\\\" \\\"MDU6SXNzdWUxMTU5MTQ4MTM=\\\" \\\"MDU6SXNzdWUxMTU5MTUzOTU=\\\" \\\"MDU6SXNzdWUxMTU5MTg0NjQ=\\\" \\\"MDU6SXNzdWUxMTU5MjIxMzU=\\\" \\\"MDU6SXNzdWUxMTU5MjU5MjM=\\\" \\\"MDU6SXNzdWUxMTU5MjU5NjQ=\\\" \\\"MDU6SXNzdWUxMTU5MjYyMDg=\\\" \\\"MDU6SXNzdWUxMTU5MjYzNjI=\\\" \\\"MDU6SXNzdWUxMTU5Mjc3NjI=\\\" \\\"MDU6SXNzdWUxMTU5Mjc5ODc=\\\" \\\"MDU6SXNzdWUxMTU5MjgwOTc=\\\" \\\"MDU6SXNzdWUxMTU5Mjg4MzI=\\\" \\\"MDU6SXNzdWUxMTU5MzI2MDc=\\\" \\\"MDU6SXNzdWUxMTU5MzgyNDg=\\\" \\\"MDU6SXNzdWUxMTU5MzkwMjE=\\\" \\\"MDU6SXNzdWUxMTU5NDA5ODQ=\\\" \\\"MDU6SXNzdWUxMTU5NDMzNTc=\\\" \\\"MDU6SXNzdWUxMTU5NDM5NTc=\\\" \\\"MDU6SXNzdWUxMTU5NDQzNzM=\\\" \\\"MDU6SXNzdWUxMTU5NDQ4NjE=\\\" \\\"MDU6SXNzdWUxMTU5NDc5NDg=\\\" \\\"MDU6SXNzdWUxMTU5NDgxNDg=\\\" \\\"MDU6SXNzdWUxMTU5NDg4Nzc=\\\" \\\"MDU6SXNzdWUxMTU5NTIyNzI=\\\" \\\"MDU6SXNzdWUxMTU5NTU1NDc=\\\" \\\"MDU6SXNzdWUxMTU5NTY5Mjc=\\\" \\\"MDU6SXNzdWUxMTU5NjUwMjA=\\\" \\\"MDU6SXNzdWUxMTU5NjYxNDc=\\\" \\\"MDU6SXNzdWUxMTU5Njk3Mjc=\\\" \\\"MDU6SXNzdWUxMTU5NzA2OTk=\\\" \\\"MDU6SXNzdWUxMTU5NzM0MDE=\\\" \\\"MDU6SXNzdWUxMTU5ODE1NzU=\\\" \\\"MDU6SXNzdWUxMTU5ODU1NTI=\\\" \\\"MDU6SXNzdWUxMTU5ODYxNzE=\\\" \\\"MDU6SXNzdWUxMTU5OTE5MDE=\\\" \\\"MDU6SXNzdWUxMTU5OTI3MjI=\\\" \\\"MDU6SXNzdWUxMTU5OTQyMzg=\\\" \\\"MDU6SXNzdWUxMTU5OTY5MTQ=\\\" \\\"MDU6SXNzdWUxMTU5OTY5NTc=\\\" \\\"MDU6SXNzdWUxMTU5OTc1NTU=\\\" \\\"MDU6SXNzdWUxMTU5OTg4MDA=\\\" \\\"MDU6SXNzdWUxMTYwMDAwNTI=\\\" \\\"MDU6SXNzdWUxMTYwMDEwMzk=\\\" \\\"MDU6SXNzdWUxMTYwMDEzMDg=\\\" \\\"MDU6SXNzdWUxMTYwMDE5OTU=\\\" \\\"MDU6SXNzdWUxMTYwMDU3NTA=\\\" \\\"MDU6SXNzdWUxMTYwMDY5MjA=\\\" \\\"MDU6SXNzdWUxMTYwMDczMTQ=\\\" \\\"MDU6SXNzdWUxMTYwMDc2MDU=\\\" \\\"MDU6SXNzdWUxMTYwMDgwODU=\\\" \\\"MDU6SXNzdWUxMTYwMDkzNjQ=\\\" \\\"MDU6SXNzdWUxMTYwMTA1NDg=\\\" \\\"MDU6SXNzdWUxMTYwMTE1MDE=\\\" \\\"MDU6SXNzdWUxMTYwMTM5MzY=\\\" \\\"MDU6SXNzdWUxMTYwMTgwNzk=\\\" \\\"MDU6SXNzdWUxMTYwMTg0OTI=\\\" \\\"MDU6SXNzdWUxMTYwMTk1OTA=\\\" \\\"MDU6SXNzdWUxMTYwMjAwNTU=\\\" \\\"MDU6SXNzdWUxMTYwMjE3NTY=\\\" \\\"MDU6SXNzdWUxMTYwMjc2OTY=\\\" \\\"MDU6SXNzdWUxMTYwMjgzOTU=\\\" \\\"MDU6SXNzdWUxMTYwMjg5ODQ=\\\" \\\"MDU6SXNzdWUxMTYwMzY2NDk=\\\" \\\"MDU6SXNzdWUxMTYwNDgyODU=\\\" \\\"MDU6SXNzdWUxMTYwNTM3NTM=\\\" \\\"MDU6SXNzdWUxMTYwNTM3NTg=\\\" \\\"MDU6SXNzdWUxMTYwNTY1OTA=\\\" \\\"MDU6SXNzdWUxMTYwNTgwMDk=\\\" \\\"MDU6SXNzdWUxMTYwNTg3NTg=\\\" \\\"MDU6SXNzdWUxMTYwNjEwNzc=\\\" \\\"MDU6SXNzdWUxMTYwNjMwMzc=\\\" \\\"MDU6SXNzdWUxMTYwNjM4Njg=\\\" \\\"MDU6SXNzdWUxMTYwODA4Mjg=\\\" \\\"MDU6SXNzdWUxMTYwODU0OTg=\\\" \\\"MDU6SXNzdWUxMTYwOTY5NDY=\\\" \\\"MDU6SXNzdWUxMTYwOTc0OTc=\\\" \\\"MDU6SXNzdWUxMTYxMTg1MTU=\\\" \\\"MDU6SXNzdWUxMTYxMjc3NDA=\\\" \\\"MDU6SXNzdWUxMTYxMjk5MjM=\\\" \\\"MDU6SXNzdWUxMTYxMzkyMzc=\\\" \\\"MDU6SXNzdWUxMTYxMzk5MTc=\\\" \\\"MDU6SXNzdWUxMTYxNDU1MzE=\\\" \\\"MDU6SXNzdWUxMTYxNTUyNzY=\\\" \\\"MDU6SXNzdWUxMTYxNjI5OTk=\\\" \\\"MDU6SXNzdWUxMTYxOTg2NzA=\\\" \\\"MDU6SXNzdWUxMTYyMDU0MjU=\\\" \\\"MDU6SXNzdWUxMTYyMTY2MjI=\\\" \\\"MDU6SXNzdWUxMTYyMjQ1NDU=\\\" ]) { ... on Issue { __typename id url title body number author { __typename ... on User { id } ... on Bot { id } ... on Organization { id } } } } node_0: node(id: \\\"MDEwOlJlcG9zaXRvcnk0NTcxNzI1MA==\\\") { id ... on Repository { issues(first: 100 after: \\\"Y3Vyc29yOnYyOpHOBu2L1A==\\\") { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } pullRequests(first: 100 after: \\\"Y3Vyc29yOnYyOpHOA1GVSA==\\\") { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_1: node(id: \\\"MDU6SXNzdWUxMTU4ODYzMDI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_2: node(id: \\\"MDU6SXNzdWUxMTU4OTQxMzg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_3: node(id: \\\"MDU6SXNzdWUxMTU4OTY2NTY=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_4: node(id: \\\"MDU6SXNzdWUxMTU4OTg0NDk=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_5: node(id: \\\"MDU6SXNzdWUxMTU5MTA5MDA=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_6: node(id: \\\"MDU6SXNzdWUxMTU5MTE3MzU=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_7: node(id: \\\"MDU6SXNzdWUxMTU5MTIxMTE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_8: node(id: \\\"MDU6SXNzdWUxMTU5MTQ0OTU=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_9: node(id: \\\"MDU6SXNzdWUxMTU5MTQ2OTE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_10: node(id: \\\"MDU6SXNzdWUxMTU5MTQ3NjY=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_11: node(id: \\\"MDU6SXNzdWUxMTU5MTQ4MTM=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_12: node(id: \\\"MDU6SXNzdWUxMTU5MTUzOTU=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_13: node(id: \\\"MDU6SXNzdWUxMTU5MTg0NjQ=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_14: node(id: \\\"MDU6SXNzdWUxMTU5MjIxMzU=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_15: node(id: \\\"MDU6SXNzdWUxMTU5MjU5MjM=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_16: node(id: \\\"MDU6SXNzdWUxMTU5MjU5NjQ=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_17: node(id: \\\"MDU6SXNzdWUxMTU5MjYyMDg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_18: node(id: \\\"MDU6SXNzdWUxMTU5MjYzNjI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_19: node(id: \\\"MDU6SXNzdWUxMTU5Mjc3NjI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_20: node(id: \\\"MDU6SXNzdWUxMTU5Mjc5ODc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_21: node(id: \\\"MDU6SXNzdWUxMTU5MjgwOTc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_22: node(id: \\\"MDU6SXNzdWUxMTU5Mjg4MzI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_23: node(id: \\\"MDU6SXNzdWUxMTU5MzI2MDc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_24: node(id: \\\"MDU6SXNzdWUxMTU5MzgyNDg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_25: node(id: \\\"MDU6SXNzdWUxMTU5MzkwMjE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_26: node(id: \\\"MDU6SXNzdWUxMTU5NDA5ODQ=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_27: node(id: \\\"MDU6SXNzdWUxMTU5NDMzNTc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_28: node(id: \\\"MDU6SXNzdWUxMTU5NDM5NTc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_29: node(id: \\\"MDU6SXNzdWUxMTU5NDQzNzM=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_30: node(id: \\\"MDU6SXNzdWUxMTU5NDQ4NjE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_31: node(id: \\\"MDU6SXNzdWUxMTU5NDc5NDg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_32: node(id: \\\"MDU6SXNzdWUxMTU5NDgxNDg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_33: node(id: \\\"MDU6SXNzdWUxMTU5NDg4Nzc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_34: node(id: \\\"MDU6SXNzdWUxMTU5NTIyNzI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_35: node(id: \\\"MDU6SXNzdWUxMTU5NTU1NDc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_36: node(id: \\\"MDU6SXNzdWUxMTU5NTY5Mjc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_37: node(id: \\\"MDU6SXNzdWUxMTU5NjUwMjA=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_38: node(id: \\\"MDU6SXNzdWUxMTU5NjYxNDc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_39: node(id: \\\"MDU6SXNzdWUxMTU5Njk3Mjc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_40: node(id: \\\"MDU6SXNzdWUxMTU5NzA2OTk=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_41: node(id: \\\"MDU6SXNzdWUxMTU5NzM0MDE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_42: node(id: \\\"MDU6SXNzdWUxMTU5ODE1NzU=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_43: node(id: \\\"MDU6SXNzdWUxMTU5ODU1NTI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_44: node(id: \\\"MDU6SXNzdWUxMTU5ODYxNzE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_45: node(id: \\\"MDU6SXNzdWUxMTU5OTE5MDE=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_46: node(id: \\\"MDU6SXNzdWUxMTU5OTI3MjI=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_47: node(id: \\\"MDU6SXNzdWUxMTU5OTQyMzg=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_48: node(id: \\\"MDU6SXNzdWUxMTU5OTY5MTQ=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } node_49: node(id: \\\"MDU6SXNzdWUxMTU5OTY5NTc=\\\") { id ... on Issue { comments(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } reactions(first: 100) { totalCount pageInfo { endCursor hasNextPage } nodes { __typename id } } } } }\",\"variables\":{}}","error":{"data":null,"errors":[{"message":"Something went wrong while executing your query. Please include `90A8:0A08:2363474:2E3BC08:5C118CDC` when reporting this issue."}]}}
    {"type":"GRAPHQL_ERROR","retry":false,"error":{"data":null,"errors":[{"message":"Something went wrong while executing your query. Please include `90A8:0A08:2363474:2E3BC08:5C118CDC` when reporting this issue."}]}}

**Comment by decentralion** [comment]
Ok, I've dug into this a bit more. As I suspected, it's another instance of #996, specifically the case where GitHub inconsistently reports an organization as a user. The specific piece of the query that goes wrong is:

```
  owndata_1: nodes(ids: [ "MDQ6VXNlcjY2MDkyODk=" "MDQ6VXNlcjUxNzcyMzk=" "MDQ6VXNlcjc3NTM5NTk=" "MDQ6VXNlcjgzMzA4OTY=" "MDQ6VXNlcjI5ODE3NjY=" "MDQ6VXNlcjI4Mzg1NTE=" "MDQ6VXNlcjEwOTQxODA2" "MDQ6VXNlcjc1ODU2OA==" "MDQ6VXNlcjc5MDU3" "MDQ6VXNlcjg0MDA5MDc=" "MDQ6VXNlcjEwMjcyMTE5" "MDQ6VXNlcjUxNTUxMDU=" "MDQ6VXNlcjEzOTg2ODQ=" "MDQ6VXNlcjM0ODc0MDk=" "MDQ6VXNlcjg5NjAyOTY=" "MDQ6VXNlcjgzMDk0NjU=" "MDQ6VXNlcjE4MzQ0Mjgz" "MDQ6VXNlcjM1ODQ0MzI=" "MDQ6VXNlcjEyODgzNTc0" "MDQ6VXNlcjEwNzAzMjE=" "MDQ6VXNlcjE2NTQxMjQ4" "MDQ6VXNlcjE0OTI4MDI=" "MDQ6VXNlcjUwMzE2OA==" "MDQ6VXNlcjQ5MDc3NDQ=" "MDQ6VXNlcjQ3MjA4Mw==" "MDQ6VXNlcjQwMTAzOTE=" "MDQ6VXNlcjUwNzA2" "MDQ6VXNlcjUzMjUxMTI=" "MDQ6VXNlcjI3NTIxMTg=" "MDQ6VXNlcjE3ODA5NzA=" "MDQ6VXNlcjEzNjg0NjYx" "MDQ6VXNlcjExMzczOTM0" "MDQ6VXNlcjIzMDA5Mzky" "MDQ6VXNlcjU3NDY0MDk=" "MDQ6VXNlcjY4MDg5NDY=" "MDQ6VXNlcjM2MDU0" "MDQ6VXNlcjEyMzAxMg==" "MDQ6VXNlcjc5ODAyODQ=" "MDQ6VXNlcjM5MjY2ODI=" "MDQ6VXNlcjQ0NTczMw==" "MDQ6VXNlcjQxMzMwMA==" "MDQ6VXNlcjQ2NDEwMDk=" "MDQ6VXNlcjgyNzEyOTE=" "MDQ6VXNlcjQ2Mzk5NDQ=" "MDEyOk9yZ2FuaXphdGlvbjEwNTQ2NzYx" "MDQ6VXNlcjE2NjQxNTcw" "MDQ6VXNlcjcyNDQzMTc=" "MDQ6VXNlcjYxMTQ3MA==" "MDQ6VXNlcjMzMzE5Mjk=" "MDQ6VXNlcjM1MDIxMw==" "MDQ6VXNlcjE2MDczMzU5" "MDQ6VXNlcjM5OTcwMDc=" "MDQ6VXNlcjE0MTA5OTA3" "MDQ6VXNlcjEyNDExMTE0" "MDQ6VXNlcjExMjg4NjM=" "MDQ6VXNlcjExNjU3ODYw" "MDQ6VXNlcjI5NTk0MTI=" "MDQ6VXNlcjM3MTk3MA==" "MDQ6VXNlcjExNDM3Njc4" "MDQ6VXNlcjE4NjU0NzM4" "MDQ6VXNlcjk0MDMwNTc=" "MDQ6VXNlcjYwNjU2NQ==" "MDQ6VXNlcjE2MTQ1OQ==" "MDQ6VXNlcjgw" "MDQ6VXNlcjQ1ODY3Njk=" "MDQ6VXNlcjcxNjI0NTg=" "MDQ6VXNlcjEzMzg1MTkx" "MDQ6VXNlcjQyOTI1Mg==" "MDQ6VXNlcjE4NTE1MTkw" "MDQ6VXNlcjEwNjI2Mzk4" "MDQ6VXNlcjkxNjIyOTQ=" "MDQ6VXNlcjIzNDcwMzcz" "MDQ6VXNlcjE2MzE4NjI=" ]) {
    ... on User {
      __typename
      id
      url
      login
    }
  }
```

Note how the query is only anticipating Users to come back. But if we actually ask the GitHub explorer...

# query
```
{
  node(id: "MDEyOk9yZ2FuaXphdGlvbjEwNTQ2NzYx=") {
    __typename
    ...on Organization {
      login
    }
  }
}
```

# response
```
{
  "data": {
    "node": {
      "__typename": "Organization",
      "login": "factai"
    }
  }
}
```

Then GItHub reports that it's an organization (factai).

Fortunately, we already have a proposal from @wchargin to fix this entire class of issue, over at #998. It's relatively involved, and definitely the best path forward.

In the mean-time, there are ways to more easily unblock you @Dim25 if you want to run SourceCred on these repositories. Right now the error is relatively hard to track down because the error message doesn't print the ids of the offending nodes. If it did, then we could simply update the blacklisted id set with the bad ids (like the organization factai) and since conversions between users->bots and users->orgs is pretty rare, we would likely just need to blacklist a few ids to get SourceCred working on TensorFlow and Keras.

Specifically, when we get back the entries corresponding to the query posted in the mirror module, we don't anticipate that the we may have requested the wrong type of node (because we assumed, incorrectly, that GitHub gives us accurate __typenames). So `entry` is the empty object and we throw an error here:

https://github.com/sourcecred/sourcecred/blob/bbe773bb67ef6132554335c643d676fb9d4fef9d/src/graphql/mirror.js#L1220-L1226

Once we're in that function, it's a little late to print useful information, because we no longer have the original query, so we can't figure out which id corresponds to the empty entry. We could potentially go back a few stack frames, though, and apply validation here:

https://github.com/sourcecred/sourcecred/blob/bbe773bb67ef6132554335c643d676fb9d4fef9d/src/graphql/mirror.js#L753-L758

At that point, we have both the query and the response, so we could iterate over every entry in the response and, if it is empty (or doesn't have an id?), we could throw an error with the corresponding id in the posted query.

Alternatively, we could try to change the originally posted query so that it gets the `id` unconditionally, rather than conditionally on the node being a user. However, I'm not sure how involved this change will be.

**Comment by decentralion** [comment]
This should be fixed now. If it re-occurs, please open a new issue.
